Source/Scene/Cesium3DTilesetTraversal.js

```diff
@@ -1,856 +1,671 @@
-            return emptyArray;
-        }
+        var error;
+        if (frameState.mode === SceneMode.SCENE2D || frustum instanceof OrthographicFrustum) {
+            if (defined(frustum._offCenterFrustum)) {
+                frustum = frustum._offCenterFrustum;
+            }
+            var width = context.drawingBufferWidth;
+            var pixelSize = Math.max(frustum.top - frustum.bottom, frustum.right - frustum.left) / Math.max(width, height);
+            error = geometricError / pixelSize;
+        } else {
+            var distance = Math.max(tile._distanceToCamera, CesiumMath.EPSILON7);
+            var sseDenominator = camera.frustum.sseDenominator;
+            error = (geometricError * height) / (distance * sseDenominator);
 
-        var children = tile.children;
-        var childrenLength = children.length;
-        for (var i = 0; i < childrenLength; ++i) {
-            var child = children[i];
-            loadTile(tileset, child, frameState, true);
-            touch(tileset, child, outOfCore);
-            if (!tile.contentAvailable) {
-                this.allLoaded = false;
+            if (tileset.dynamicScreenSpaceError) {
+                var density = tileset._dynamicScreenSpaceErrorComputedDensity;
+                var factor = tileset.dynamicScreenSpaceErrorFactor;
+                var dynamicError = CesiumMath.fog(distance, density) * factor;
+                error -= dynamicError;
             }
         }
-        return children;
-    };
 
-    InternalBaseTraversal.prototype.updateAndCheckChildren = BaseTraversal.prototype.updateAndCheckChildren;
-
-    function SkipTraversal(options) {
-        this.tileset = undefined;
-        this.frameState = undefined;
-        this.outOfCore = undefined;
-        this.queue1 = new ManagedArray();
-        this.queue2 = new ManagedArray();
-        this.internalDFS = new InternalSkipTraversal(options.selectionHeuristic);
-        this.maxChildrenLength = 0;
-        this.scratchQueue = new ManagedArray();
-    }
-
-    SkipTraversal.prototype.execute = function(tileset, root, frameState, outOfCore) {
-        this.tileset = tileset;
-        this.frameState = frameState;
-        this.outOfCore = outOfCore;
-        this.internalDFS.frameState = frameState;
-        this.internalDFS.outOfCore = outOfCore;
-
-        this.maxChildrenLength = 0;
-        breadthFirstSearch(root, this);
-        this.queue1.length = 0;
-        this.queue2.length = 0;
-        this.scratchQueue.length = 0;
-        this.scratchQueue.trim(this.maxChildrenLength);
-    };
+        return error;
+    }
 
-    SkipTraversal.prototype.visitStart = function(tile) {
-        if (!isVisited(tile, this.frameState)) {
-            visitTile(this.tileset, tile, this.frameState, this.outOfCore);
+    function updateVisibility(tileset, tile, frameState) {
+        if (tile._updatedVisibilityFrame === frameState.frameNumber && !frameState.passes.pick) {
```

Comment:
Yeah I agree this wasn't very obvious. I updated the code so that `_updatedVisibilityFrame` is split into `_updatedVisibilityFramePick` and `_updatedVisibilityFrameRender`.

The reason `_updatedVisibilityFrame` exists at all is so we don't check visibility twice when the `cullWithChildrenBounds` optimization is used. But I want to think of way so that we don't need this flag.